File: 2.hashmap/290. Word Pattern.py
def chekLen(patternLen):
    counter = 0
    for i in myString:
        if i == ' ':
            counter += 1
    return counter+1
        
def getMyDict():
    myDict = {}
    addString = ''
    up = 0
  
    for i in myString:
            
            print(pattern[up])
            if i != ' ':
                addString += i
                myDict[pattern[up]] = addString

            if i == ' ':
                addString = ''
                up += 1
    return myDict
    
def wordPattern(pattern, myString):
    patternLen = len(pattern)
    myStringLen = len(myString)
    
    if not patternLen == chekLen(patternLen):return False
    myDict = getMyDict()
    print(myString.split())


pattern = "abcd"
myString = "dog cat cat fish"
(wordPattern(pattern, myString))

File: javascript/290. Word Pattern.js
function worldPattern(pattern, s) {
     let dict = {};
     let j = 0;
     let seen =[];
     s = s.split(' ');
     pattern = pattern.split('');

     for (char of pattern) {
          if (!(char in dict)) {
               if (seen.includes(s[j])){return false};
               dict[char] = s[j];
               seen.push(s[j]);
          } else {
               
               if (dict[char] != s[j]){return false;}
          }
          j++;
     };
    
};

let pattern = "abba";
let s = "dog dog dog dog";
worldPattern(pattern, s);